ambry-admin/src/main/java/com.github.ambry.admin/AdminBlobStorageService.java

```diff
@@ -167,30 +170,12 @@ public void handleGet(RestRequest restRequest, RestResponseChannel restResponseC
 
   @Override
   public void handlePost(RestRequest restRequest, RestResponseChannel restResponseChannel) {
-    long processingStartTime = System.currentTimeMillis();
-    long preProcessingTime = 0;
     handlePrechecks(restRequest, restResponseChannel);
     restRequest.getMetricsTracker().injectMetrics(adminMetrics.postBlobMetrics);
-    try {
-      logger.trace("Handling POST request - {}", restRequest.getUri());
-      checkAvailable();
-      long propsBuildStartTime = System.currentTimeMillis();
-      BlobProperties blobProperties = RestUtils.buildBlobProperties(restRequest.getArgs());
-      byte[] usermetadata = RestUtils.buildUsermetadata(restRequest.getArgs());
-      adminMetrics.blobPropsBuildTimeInMs.update(System.currentTimeMillis() - propsBuildStartTime);
-      logger.trace("Blob properties of blob being POSTed - {}", blobProperties);
-      PostCallback routerCallback =
-          new PostCallback(restRequest, restResponseChannel, new BlobInfo(blobProperties, usermetadata));
-      preProcessingTime = System.currentTimeMillis() - processingStartTime;
-      SecurityProcessRequestCallback securityCallback =
-          new SecurityProcessRequestCallback(restRequest, restResponseChannel, blobProperties, usermetadata,
-              routerCallback);
-      securityService.processRequest(restRequest, securityCallback);
-    } catch (Exception e) {
-      submitResponse(restRequest, restResponseChannel, null, e);
-    } finally {
-      adminMetrics.postPreProcessingTimeInMs.update(preProcessingTime);
-    }
+    Exception exception =
```

Comment:
Since `checkAvailable()` throws the exception, I would prefer doing it this way since it returns the exception as a response to the client gracefully.